src/Path/CurrentPathStack.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Retrofit\Drupal\Path;
+
+use Drupal\Core\Path\CurrentPathStack as CoreCurrentPathStack;
+use Symfony\Component\HttpFoundation\Request;
+
+class CurrentPathStack extends CoreCurrentPathStack
+{
+    public function getPath(?Request $request = null): string
+    {
+        $request ??= $this->requestStack->getCurrentRequest();
+        assert($request instanceof Request);
+        $this->paths[$request] ??= ['path' => $request->getPathInfo()];
+        assert(is_array($this->paths[$request]) && is_string($this->paths[$request]['path']));
+        return $this->paths[$request]['path'];
+    }
```

Comment:
I'm still not 100% I understand why we're overriding `getPath` this way. Is it tied to `setPath` and allowing `$_GET['q']` to be passed and used by reference?

Reply:
That is correct. Otherwise Drupal 7 code might assume that it has changed the path when it has not.